File: app/src/main/java/com/example/jetpackcomponentsapp/utils/BottomOffsetDecorationHelper.java
package com.example.jetpackcomponentsapp.utils;

import android.content.Context;
import android.graphics.Rect;
import android.view.View;

import androidx.annotation.DimenRes;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class BottomOffsetDecorationHelper extends RecyclerView.ItemDecoration {

    private int bottomItemOffset;

    public BottomOffsetDecorationHelper(@NonNull Context context, @DimenRes int itemOffsetId) {
        bottomItemOffset = itemOffsetId;
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        int dataSize = state.getItemCount();
        int position = parent.getChildAdapterPosition(view);
        if (dataSize > 0 && position == dataSize - 1) {
            outRect.set(0,0,0,bottomItemOffset);
        } else {
          outRect.set(0,0,0,0);
        }
    }
}

File: app/src/main/java/com/example/jetpackcomponentsapp/BottomOffsetDecorationHelper.kt
package com.example.jetpackcomponentsapp

import android.content.Context
import android.graphics.Rect
import android.view.View
import androidx.annotation.DimenRes
import androidx.annotation.NonNull
import androidx.recyclerview.widget.RecyclerView

class BottomOffsetDecorationHelper : RecyclerView.ItemDecoration {

    private var bottomItemOffset: Int? = null

    constructor(@NonNull context: Context, @DimenRes itemOffsetId: Int) {
        bottomItemOffset = itemOffsetId
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        val dataSize = state.itemCount
        val position = parent.getChildAdapterPosition(view)
        if (dataSize > 0 && position == dataSize - 1) {
            outRect.set(0, 0, 0, bottomItemOffset?:0)
        } else {
            outRect.set(0, 0, 0, 0)
        }
    }
}